pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -434,65 +364,21 @@ func (am *Alertmanager) applyConfig(cfg *apimodels.PostableUserConfig, rawConfig
 	}
 
 	// With the templates persisted, create the template list using the paths.
-	tmpl, err := am.templateFromPaths(paths)
+	tmpl, err := am.Base.TemplateFromPaths(am.Settings.AppURL, paths...)
 	if err != nil {
 		return err
 	}
 
-	// Finally, build the integrations map using the receiver configuration and templates.
-	integrationsMap, err := am.buildIntegrationsMap(cfg.AlertmanagerConfig.Receivers, tmpl)
+	err = am.Base.ApplyConfig(AlertingConfiguration{
```

Comment:
This diff is somewhat clear but the gist is that everything where we doing is now done by `Base.ApplyConfig` - whatever we need from here - we inject via `AlertingConfiguration`